src/Loop.php

```diff
@@ -42,12 +42,11 @@ public static function setFactory(DriverFactory $factory = null)
         }
 
         self::$factory = $factory;
+        self::$driver = null; // reset it here, it will be actually instantiated inside execute() or get()
 
         if ($factory === null) {
-            self::$driver = null;
             self::$registry = null;
         } else {
-            self::$driver = self::createDriver();
             self::$registry = [];
```

Comment:
Yes, sure. But the only reason to do so is `wait`. Anything other shouldn't reuse a loop. If we didn't have wait, we could explicitly forbid that. Maybe leave it as trait and make it a must for `LoopDriver`?